src/libraries/System.Private.CoreLib/src/System/IO/Path.Windows.cs

```diff
@@ -164,6 +179,27 @@ private static void GetTempPath(ref ValueStringBuilder builder)
                 throw Win32Marshal.GetExceptionForLastWin32Error();
 
             builder.Length = (int)result;
+
+            static uint GetTempPathW(int bufferLen, ref char buffer)
+            {
+                delegate* unmanaged<int, char*, uint> func = s_GetTempPathWFunc;
+                if (func == null)
+                {
+                    func = s_GetTempPathWFunc = GetGetTempPathWFunc();
+                }
+
+                int lastError;
+                uint retVal;
+                fixed (char* ptr = &buffer)
+                {
+                    Marshal.SetLastSystemError(0);
```

Comment:
Why you need to `SetLastSystemError(0)`?

Reply:
That's what the source-generated marshalling code does.

Reply:
Presumably to avoid random behavior if calling a function that does not set the last error, and you nevertheless read it. @elinor-fung

Reply:
Although if e we trust the documentation of both functions, it is not necessary.